python_modules/dagster/dagster/_core/definitions/asset_condition.py

```diff
@@ -175,65 +217,54 @@ def with_run_ids(self, run_ids: AbstractSet[str]) -> "AssetConditionEvaluationWi
         return AssetConditionEvaluationWithRunIds(evaluation=self, run_ids=frozenset(run_ids))
 
 
-class AssetConditionEvaluationResult(NamedTuple):
-    """Return value for the evaluate method of an AssetCondition."""
+@whitelist_for_serdes
+class AssetConditionEvaluationState(NamedTuple):
+    """Incremental state calculated during the evaluation of an AssetCondition. This may be used
+    on the subsequent evaluation to make the computation more efficient.
+    """
 
-    condition: "AssetCondition"
     evaluation: AssetConditionEvaluation
```

Comment:
Thoughts on calling this `latest_evaluation` (with implications for some of the other properties)?

Reply:
Yeah I've been going back and forth on that point -- my feeling was that it might be redundant, as we can just call the entire object `previous_evaluation_state`. If we do that, then `previous_evaluation_state.previous_evaluation` is a bit weird.

Reply:
The way I am inclined to look at it is that the "state" isn't exactly the state _of_ the previous valuation. It's more that the ongoing incremental state we track includes the previous evaluation.

E.g. imagine a world where we want to get smart about the order that we compute rules in, and, to do that, we want to track statistics on the typical amount of time it takes to evaluate conditions. To avoid overfitting on the previous tick, we might want to track the last N ticks.  That data could make sense to store inside this object, but wouldn't be part of the previous evaluation.

Reply:
ok updated it to `previous_evaluation` and `previous_tick_evaluation_timestamp`. kept "max_storage_id" as is, as "previous_max_storage_id" feels like the max storage id that was used for the previous evaluation, which is one level "too deep"